nixos/modules/services/matrix/conduwuit.nix

```diff
@@ -0,0 +1,299 @@
+
+              To listen on multiple addresses, specify a list. For example:
+              `[ "127.0.0.1" "::1" ]`
+            '';
+          };
+          global.port = lib.mkOption {
+            type = lib.types.oneOf [
+              lib.types.port
+              (lib.types.listOf lib.types.port)
+            ];
+            default = 6167;
+            description = "The port(s) conduwuit will be running on. You need to set up a reverse proxy in your web server (e.g. apache or nginx), so all requests to /_matrix on port 443 and 8448 will be forwarded to the conduwuit instance running on this port";
+          };
+          global.unix_socket_path = lib.mkOption {
+            type = lib.types.nullOr lib.types.path;
+            default = null;
+            description = ''
+              Listen on a UNIX socket at the specified path. If listening on a UNIX socket,
+              listening on an address will be disabled. The `address` option must be set to
+              `null` (the default value). You must add your reverse proxy to the `conduwuit`
+              group or set {option}`services.conduwuit.group` to a group your reverse proxy
+              is part of.
+
+              This will automatically add a static user to your system if {option}`services.conduwuit.user`
+              is left at the default value `conduwuit, and a static group if {option}`services.conduwuit.group`
+              is left at the default value `conduwuit`, since this is the intended way how
+              unix sockets should be used with conduwuit.
+            '';
+          };
+          global.unix_socket_perms = lib.mkOption {
+            type = lib.types.ints.positive;
+            default = 660;
+            description = "The default permissions (in octal) to create the UNIX socket with.";
+          };
+          global.max_request_size = lib.mkOption {
+            type = lib.types.ints.positive;
+            default = 20000000;
+            description = "Max request size in bytes. Don't forget to also change it in the proxy.";
+          };
+          global.allow_registration = lib.mkOption {
+            type = lib.types.bool;
+            default = false;
+            description = ''
+              Whether new users can register on this server.
+
+              If set to true without a token configured, users can register with no
+              form of 2nd-step only if you set
+              `yes_i_am_very_very_sure_i_want_an_open_registration_server_prone_to_abuse` to
+              true in your config.
+              If you would like registration only via token reg, please configure
+              `registration_token` or `registration_token_file`.
+            '';
+          };
+          global.allow_encryption = lib.mkOption {
+            type = lib.types.bool;
+            default = true;
+            description = "Whether new encrypted rooms can be created. Note: existing rooms will continue to work.";
```

Comment:
```suggestion
            description = "Whether new encrypted rooms can be created. Existing rooms are not effected by this setting.";
```

Reply:
No language improvement in this suggestion and also it contains a grammatical error (effected -> affected).